include/model/Model.hpp

```diff
@@ -40,44 +40,16 @@ namespace model {
 /// Py_Finalize();
 class Model {
  public:
-    /// Base class for model input
-    class ModelInput;
-
-    /// Base class for model output
-    class ModelOutput;
-
-    /// Get the model's predicted output given then input
+    /// Get the output state given the action input and the current state
     ///
-    /// \param input The model input, dependent on model type
-    /// \param[out] output The ouput after running inference
-    virtual void inference(const ModelInput& input, ModelOutput* output) = 0;
-
-    /// Get the output state given the input and the current state
-    ///
-    /// \param in The in/current state
-    /// \param[out] The predicted output state given the model input
-    virtual void predict_state(
-        const ModelInput& input,
-        const aikido::statespace::StateSpace::State& in_,
-        aikido::statespace::StateSpace::State* out_) = 0;
-};
-
-/// Class for handling input of the model; varies based on model
-class Model::ModelInput {
- protected:
-    /// This is a base class that should only be used in derived classes
-    ModelInput() = default;
-
-    ~ModelInput() = default;
-};
-
-/// Class for handling output of the model; varies based on model type
-class Model::ModelOutput {
- protected:
-    /// This is a base class that should only be used in derived classes
-    ModelOutput() = default;
-
-    ~ModelOutput() = default;
+    /// \param input_action Given action vector
+    /// \param input_state Given state vector
```

Comment:
What statespace does this data belong to? What are the parameters of this vector? What's the size